django_project/analysis/models.py

```diff
@@ -765,3 +766,148 @@ class AnalysisTask(models.Model):
         blank=True,
         help_text='Error message if the task failed.'
     )
+
+
+class IndicatorSource(models.TextChoices):
+    """Choices for the source of an indicator."""
+
+    BASE = 'base', 'Base'
+    # Global Pasteur Watch
+    GPW = 'GPW', 'Global Pasteur Watch'
+    OTHER = 'other', 'Other'
+
+
+class Indicator(models.Model):
+    """Model to represent an indicator used in analysis."""
+
+    ALLOWED_ANALYSIS_TYPES = [
+        'Baseline',
+        'Temporal',
+        'Spatial'
+    ]
+    ALLOWED_TEMPORAL_RESOLUTIONS = [
+        'Annual',
+        'Quarterly',
+        'Monthly'
+    ]
+
+    name = models.CharField(
+        max_length=255,
+        unique=True,
+        help_text="The name of the indicator."
+    )
+
+    description = models.TextField(
+        blank=True,
+        null=True,
+        help_text="Description of the indicator."
+    )
+
+    variable_name = models.CharField(
+        max_length=255,
+        unique=True,
+        help_text="The variable name used in the analysis."
+    )
+
+    source = models.CharField(
+        max_length=50,
+        choices=IndicatorSource.choices,
+        default=IndicatorSource.BASE,
+        help_text="The source of the indicator."
+    )
+
+    analysis_types = models.JSONField(
+        default=list,
+        help_text="List of analysis types this indicator can be used for.",
+        blank=True,
+        null=True
+    )
+
+    temporal_resolution = models.JSONField(
```

Comment:
@danangmassandy can we change to temporal_resolutions because it will be a list?